### ISSUE TITLE
Fails to compile on Arch linux (libwlroots.so.12.p/backend_drm_libliftoff.c.o -Werror=calloc-transposed-args

### ISSUE THREAD

**Issue by mikedebian** [phkaeser/wlmaker#58]
Cloned git.

https://pastebin.com/awMKfUzU

**Comment by phkaeser** [comment]
Thank you for reporting!

This is a version mismatch in the dependency stack -- between libwlroots, libdrm and libliftoff. Could I ask you to share the full output of when you built the dependencies (https://github.com/phkaeser/wlmaker?tab=readme-ov-file#to-configure) ?

I'd want to understand which versions of libraries are used, and which ones are picked from the dependencies subtree, vs. from the host system (Arch linux).

**Comment by mikedebian** [comment]
Sure.

https://pastebin.com/E0iFM1MN

Having window maker on wayland, I'd pay for that.

**Comment by phkaeser** [comment]
Thanks! I'll need to figure out whether I can disable wlroots from using libliftoff, or may need to bump up the dependencies. => that'll take some tinkering.

Managing expectations: wlmaker is still in rather early development, not up to par with Window Maker for a bit.  (... contributions welcome, for sure :smiley: )

**Comment by mikedebian** [comment]
I know, but I've been waiting on somebody to tackle WindowMaker since Wayland became the "norm". I'm very happy it's being worked on. All progress is good progress

**Comment by phkaeser** [comment]
Looking into this => this is caused by libliftoff at 0.5.0 having changed API. 

wlmaker's main dependency is libwlroots -- this is a finicky dependency, since there's breaking API changes even on minor version changes. I had been using an older version and wouldn't mind moving to a more recent tagged version. But, the most recent version is 0.17.3, from **1 month ago**. 

There are two commits that (1) permit disabling libliftoff, and (2) conditionally fix the API, if the version is 0.5.0 or higher:

wlroots only recently added an option to disable libliftoff:
https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/385c9ade5f7a8ce9e5c13f762d56e6bd1c8d1b0a

And, support for libliftoff 0.5 was only very recently added in wlroots:
https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/6e6c4408d36ddad705458ca4b2e301192f7963fd

But, both of these were from the most recent 2 weeks, and are not contained in any tagged libwlroots version.

Options:

* I could move my dependency to libwlroots @ today. I'd prefer not doing that; I actually want to move this to the most recent tagged version (.. but, that won't fix it here). This will most likely introduce API changes throughout wlmaker.
* Consider downgrading libliftoff to a version before 0.4; or de-install it.  (Is that an option for you?)
* At last, wait until libwlroots cuts a new release. These are happening roughly every 1-2 months.

Is downgrading libliftoff an option for you? 
Otherwise I'd (unfortunately) suggest to wait until libwlroots incorporates the fix in a tagged version.

**Comment by phkaeser** [comment]
Ah, I forgot to mention: @mikedebian -- if you sync `dependencies/wlroots` to HEAD (including above-mentioned fixes), and re-configure & rebuild the dependencies from there, it should compile.

(as a stop-gap until wlroots gets bumped to 0.17.4).

**Comment by mikedebian** [comment]
I updated the libs in dependencies and it compiled. It's very pretty.
